src/common/cis2pi.jl

```diff
@@ -1,5 +1,15 @@
 """Function to compute exp(2π i x)"""
 cis2pi(x) = cispi(2x)
+function cis2pi(x::AbstractFloat)
+    # Special case when 2x is an integer, as exp(n*π*i) = +- 1. Saves expensive
+    # exponential evaluations when called repeatedly
+    if isinteger(2x)
+        isodd(2x) ? result = -Complex{typeof(x)}(1, 0) : result = Complex{typeof(x)}(1, 0)
```

Comment:
Can just use `T` here
also use `one(complex(T))`
I would write
```suggestion
        return isodd(2x) ? -one(complex(T)) : one(complex(T))
```